pkgs/by-name/cz/czkawka/package.nix

```diff
@@ -85,11 +91,14 @@ let
       install -Dm444 -t $out/share/metainfo data/com.github.qarmin.czkawka.metainfo.xml
     '';
 
+    nativeInstallCheckInputs = [
+      versionCheckHook
+    ];
+    versionCheckProgram = "${placeholder "out"}/bin/czkawka_cli";
+    versionCheckProgramArg = [ "--version" ];
+    doInstallCheck = true;
+
     passthru = {
-      tests.version = testers.testVersion {
```

Comment:
I have put it back.
Just for curiosity though, I don't see how this is not redundant.
I mean that to evaluate this derivation, the package has to have been built (i.e. `versionCheckHook` succeeds).

Hence, if I'm not mistaking, there is no knowledge difference between the state where `passthru.tests` is built and the one where it is not yet built. Am I missing something here ?

Reply:
The environment set during the check hook is not necessarily identical to the environment set during the passthru check.

This can be clearly seen by the line `versionCheckProgram = "${placeholder "out"}/bin/czkawka_cli";` you added above - further reinforced by the documentation:

> Using `$out` in the value of this variable won't work, as environment variables from this variable are not expanded by the hook. Hence using `placeholder` is unavoidable.

This is an issue that does not happen at all with the passthru check.

Therefore passthru checks are not the exact same thing as hook checks.

Reply:
Ok I see, thanks for this explanation.

Reply:
Generalizing this a bit, check hooks are more limited than passthru test derivations.

Two examples from memory:
- live555 has a passthru test that just builds VLC. Building VLC during a check hook is too cumbersome.
- handbrake has a passthru tests that fetches and reencodes Big Buck Bunny movie. Again a too cumbersome thing to be done during a check hook.

Reply:
For sure, I was not aware of the subtle differences for the specific case of version checking.